pallet-chainlink-feed/src/lib.rs

```diff
@@ -1460,8 +1412,22 @@ pub mod pallet {
 			Ok(())
 		}
 
+		/// Prune the state of a feed to reduce storage load.
+		fn prune_oldest(&mut self) {
+			let prune_next = self.config.next_round_to_prune;
+			// only prune if window is exceeded
+			if self.config.latest_round.saturating_sub(prune_next) >= self.config.pruning_window {
+				Rounds::<T>::remove(self.id, prune_next);
+				Details::<T>::remove(self.id, prune_next);
+				// update oldest round
+				self.config.next_round_to_prune += RoundId::one();
+				self.config.first_valid_round = Some(self.config.next_round_to_prune);
```

Comment:
`first_valid_round` is part of the public `FeedInterface`: 
https://github.com/ChainSafe/chainlink-polkadot/blob/c23bcae4c7d3f123afd1d1159b4f20deeeb2bff8/pallet-chainlink-feed/src/lib.rs#L1518-L1519

We need to keep shifting here so that the `first_valid_round()` getters keeps returning the first valid round